### ISSUE TITLE
[Arm64] System.Reflection.Emit.Tests.TypeBuilderIsGenericTypeDefinition.IsGenericTypeDefinition_GenericType_ReturnsTrue [FAIL]

### ISSUE THREAD

**Issue by sdmaclea** [dotnet/coreclr#18034]
This is a recent regression on tip.
```
corerun xunit.console.netcore.exe System.Reflection.Emit.Tests.dll -xml testResults.xml -notrait Benchmark=true -notrait category=nonnetcoreapptests -notrait category=nonlinuxtests -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing
xUnit.net console test runner (64-bit .NET Core)
Copyright (C) 2014 Outercurve Foundation.

Discovering: System.Reflection.Emit.Tests
Discovered:  System.Reflection.Emit.Tests
Starting:    System.Reflection.Emit.Tests
   System.Reflection.Emit.Tests.TypeBuilderIsGenericTypeDefinition.IsGenericTypeDefinition_GenericType_ReturnsTrue [FAIL]
      System.NotSupportedException : The invoked member is not supported before the type is created.
      Stack Trace:
            at System.Reflection.Emit.TypeBuilder.GetConstructorImpl(BindingFlags bindingAttr, Binder binder, CallingConventions callConvention, Type[] types, ParameterModifier[] modifiers)
            at System.Type.GetConstructor(BindingFlags bindingAttr, Binder binder, CallingConventions callConvention, Type[] types, ParameterModifier[] modifiers)
            at System.Reflection.Emit.ConstructorBuilder.GetParameters()
            at System.Reflection.Emit.ModuleBuilder.GetMemberRefToken(MethodBase method, IEnumerable`1 optionalParameterTypes)
            at System.Reflection.Emit.ModuleBuilder.GetMethodTokenInternal(MethodBase method, IEnumerable`1 optionalParameterTypes, Boolean useMethodDef)
            at System.Reflection.Emit.ILGenerator.GetMethodToken(MethodBase method, Type[] optionalParameterTypes, Boolean useMethodDef)
            at System.Reflection.Emit.ILGenerator.Emit(OpCode opcode, ConstructorInfo con)
```

**Comment by sdmaclea** [comment]
@AtsushiKan FYI

**Comment by stakx** [comment]
Yes, it indeed looks like I might have caused this. 😱 (Seems my local corefx setup is still not set up right, I should've been able to catch this in time.) This should be resolved by #18036. I'm sorry for the trouble, I'll be more careful next time around.

**Comment by stakx** [comment]
@sdmaclea - #18036 has been merged, so this regression should be resolved. Can you confirm this?

**Comment by sdmaclea** [comment]
Fixed via #18036 thanks
